### DISCUSSION TITLE
What Flightmodes does INAV use in LTM telemetry?

### DISCUSSION THREAD

**Discussion by altermac** [iNavFlight/inav#9501] [Q&A] [answered]
I'm reading LTM telemetry to display information on an external display. There seems to be a mismatch in the LTM-documentation of INAV, LTM telemetry flightmodes seem to be different than CLI AUX modes. So far i got the following list (c++) by testing 'on the bench': 

// Flightmodes
static const char* flightModes[] = {
  "Manual", // inav checked = 0
  "Acro", // inav checked = 1
  "Angle", // inav checked = 2
  "Horizon", // inav checked = 3
  "Acro",
  "Stabilized1",
  "Stabilized2",
  "Stabilized3",
  "Altitude Hold",
  "GPS Hold",
  "Waypoints",
  "Head free",
  "Circle",
  "RTH",
  "Follow me",
  "Land",
  "Fly by wire A",
  "Fly by wire B",
  "Cruise",
  "Unknown"
};

List starts on zero: Manual=0, Acro=1,.... 
Since LTM standard  is shared with betaflight/cleanflight/ardupilot/INAV, my list seems to be from another software, maybe ardupilot.

Can anyone tell me the complete list for INAV or lead me to the place where LTM flightmodes are set in inav sourcecode?

Thanks in advance.

**Comment by stronnag** [comment]
LTM is documented on the INAV wiki, for example, [status frame](https://github.com/iNavFlight/inav/wiki/Lightweight-Telemetry-(LTM)#status-frame-s)

**Reply by altermac** [reply]
There seem to be differences between documentation and LTM-output. It's a bit difficult for me to force Navigation modes 'on the bench'. I would like to use/translate the names of the mode as seen in INAV configurator (AUX).

**Comment by stronnag** [comment, marked as answer]
> There seem to be differences between documentation and LTM-output. It's a bit difficult for me to force Navigation modes 'on the bench'. I would like to use/translate the names of the mode as seen in INAV configurator (AUX).

I don't believe there are any issues; if I use the SITL to replay a blackbox using `fl2stil` (so modes asserted by  generated RC commands), and view the SITL's resulting LTM in `mwp`; then the visualised modes from LTM appear to be correct.

But please raise an issue if you do discover any anomalies.

**Comment by altermac** [comment]
Thanks for your input, problem solved:
Found the coding for LTM flightmodes frame in GitHub\inav\src\main\telemetry\LTM.c. Seems that only a handfull of modes is supported by INAV.
